src/route.js

```diff
@@ -27,11 +29,24 @@ var Route = function (routes, uri) {
 Route.prototype = {
 
   /**
+  Attempt to match the uri, or call a notFound handler if nothing matches.
+
+  @method matchOrNotFound
+  @param {Object} routeLevel
+  **/
+  matchOrNotFound: function (routeLevel) {
+    this.match(routeLevel);
+
+    if (!this.fullMatch && this.notFound) {
```

Comment:
Spoke with @nahiluhmot in person. We think this case is actually OK because the not found action would only be called after all matched `/*` actions. And if you had a `/*` action that redirected based on permissions or what have you, that call to `navigate` halts all later actions.